crates/cranelift/src/wasmfx/optimized.rs

```diff
@@ -1031,28 +1203,53 @@ pub(crate) fn translate_resume<'a>(
             );
         }
 
-        // Make the currently running continuation the parent of the one we are about to resume.
-        let original_running_contobj = typed_continuations_load_continuation_object(env, builder);
-        typed_continuations_store_parent(env, builder, original_contobj, original_running_contobj);
+        // Make the currently running continuation (if any) the parent of the one we are about to resume.
+        let original_stack_chain =
+            tc::VMContext::new(vmctx, env.pointer_type()).load_stack_chain(env, builder);
+        let original_stack_chain_raw_parts = original_stack_chain.to_raw_parts();
+        original_stack_chain.assert_not_absent(env, builder);
+        tc::ContinuationObject::new(original_contobj, env.pointer_type()).set_parent_stack_chain(
+            env,
+            builder,
+            &original_stack_chain,
+        );
 
-        builder.ins().jump(resume_block, &[original_contobj]);
-    }
+        builder.ins().jump(
+            resume_block,
+            &[
+                original_contobj,
+                original_stack_chain_raw_parts[0],
+                original_stack_chain_raw_parts[1],
+            ],
+        );
+    };
 
     // Resume block: actually resume the fiber corresponding to the
     // continuation object given as a parameter to the block. This
     // parameterisation is necessary to enable forwarding, requiring us
     // to resume objects other than `original_contobj`.
     // We make the continuation object that was actually resumed available via
```

Comment:
Currently, I'd think `original_contobj` and `resumed_contobj` are always the same object. As forwarding is essentially equivalent to `resume(suspend())`, where `resume` is bound to the continuation of the child fiber.

Reply:
The parent of `resumed_contobj` is always the same, because it is parent that performs the forwarding `suspend()`. However, the parent of the parent can change, but that's no concern here.

Reply:
Can we get rid of the redundant `suspend_block` too, now that we are at it?

Reply:
#99 will add instructions to that block